### PR TITLE
common, server: Use HookTube JSON API and add error function

### PR DESCRIPTION
**Pull request by Chiiruno**
This should be a bit lighter on the server, and covers an error case in which it was possible for the title to not fetch on the /embed hooktube url.

### CODE REVIEW

File: go/src/meguca/server/router.go [hunk 1]
```diff
@@ -250,13 +251,57 @@ func healthCheck(w http.ResponseWriter, r *http.Request) {
...
+		}
+
+		if err != nil {
+			httpError(w, r, err)
+			return
+		}
+	}
+
+	defer resp.Body.Close()
+	buf := new(bytes.Buffer)
```
Comment (bakape) [inline go/src/meguca/server/router.go]: Can be stack allocated (`var buf bytes.Buffer`) . No need for inderection.

File: go/src/meguca/common/errors.go [hunk 2]
```diff
@@ -82,6 +82,11 @@ func ErrInvalidBoard(board string) error {
 	return StatusError{fmt.Errorf("board `%s` does not exist", board), 404}
 }
 
+// Missing string
+func ErrMissingString(str string) error {
```
Comment (bakape) [inline go/src/meguca/common/errors.go]: No need for an extra public function, if you only use it once. Just inline the StatusError creation in getHookTubeTitle().

**Comment by Chiiruno** [comment]
image

hooktube is deprecated lol

**Comment by Chiiruno** [comment]
Also I tried to push my commit, but github seems to be kill.

**Comment by Chiiruno** [comment]
I could have meguca act as youtube-dl, or maybe even the client, and use that instead of hooktube, and with the added benefit of not having youtube botnet JS run on meguca's page.

**Comment by Chiiruno** [comment]
Er, not downloading the whole video, just getting the googlevideo link.

**Comment by Chiiruno** [comment]
Strangely enough, if you go to my meguca fork and go to the dev/hooktube branch, it shows the commit, but not here.

**Comment by bakape** [comment]
Maybe revert to noembed then?

**Comment by Chiiruno** [comment]
That will require some work, right now the code in this PR works, so I'd rather dedicate my time to #793

**Comment by Chiiruno** [comment]
Also travis fucked up, works on my machine.

**Comment by bakape** [comment]
I will reject this and revert back to noembed myself then.

**Comment by Chiiruno** [comment]
@bakape I'm already more than half-way done implementing hooktube-like functionality, so I guess don't revert, so I can avoid a PR conflict please.

**Comment by bakape** [comment]
Okay.

**Pull request closed without merging**